ci-operator/step-registry/upi/install/heterogeneous/powervs/upi-install-heterogeneous-powervs-commands.sh

```diff
@@ -0,0 +1,156 @@
+function wait_for_nodes_readiness()
+{
+  local expected_nodes=${1}
+  local max_retries=${2:-10}
+  local period=${3:-5}
+  for i in $(seq 1 "${max_retries}") max; do
+    if [ "${i}" == "max" ]; then
+      echo "[ERROR] Timeout reached. ${expected_nodes} ready nodes expected, found ${ready_nodes}... Failing."
+      return 1
+    fi
+    sleep "${period}m"
+    ready_nodes=$(get_ready_nodes_count)
+    if [ x"${ready_nodes}" == x"${expected_nodes}" ]; then
+        echo "[INFO] Found ${ready_nodes}/${expected_nodes} ready nodes, continuing..."
+        return 0
+    fi
+    echo "[INFO] - ${expected_nodes} ready nodes expected, found ${ready_nodes}..." \
+      "Waiting ${period}min before retrying (timeout in $(( (max_retries - i) * (period) ))min)..."
+  done
+}
+
+EXPECTED_NODES=$(( $(get_ready_nodes_count) + ADDITIONAL_WORKERS ))
+
+echo "Cluster type is ${CLUSTER_TYPE}"
+
+case "$CLUSTER_TYPE" in
+*ibmcloud*)
+  # Add code for ppc64le
+  if [ "${ADDITIONAL_WORKER_ARCHITECTURE}" == "ppc64le" ]; then
+      echo "Adding additional ppc64le nodes"
+      IBMCLOUD_HOME_FOLDER=/tmp/ibmcloud
+      SERVICE_NAME=power-iaas
+      SERVICE_PLAN_NAME=power-virtual-server-group
+      WORKSPACE_NAME=rdr-mac-$REGION-n1 # TODO: Should this be an env variable or some randomly generated name based on the zone?
+      OPENSHIFT_CLIENT_TARBALL=""
+
+      mkdir -p ${IBMCLOUD_HOME_FOLDER}
+      if [ -z "$(command -v ibmcloud)" ]; then
+        echo "ibmcloud CLI doesn't exist, installing"
+        curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+      fi
+
+      function ic() {
+        HOME=${IBMCLOUD_HOME_FOLDER} ibmcloud "$@"
+      }
+
+      ic version
+      ic login --apikey @${CLUSTER_PROFILE_DIR}/ibmcloud-api-key -r ${REGION}
+      ic plugin install -f cloud-internet-services vpc-infrastructure cloud-object-storage power-iaas
+
+      # create workspace for power from cli
+      echo "Display all the variable values"
+      echo "Region is ${REGION} Zone is ${ZONE} Resource Group is ${RESOURCE_GROUP}" #TODO: rename to RESOURCE_GROUP
+      SERVICE_INSTANCE_OUTPUT=$(ic resource service-instance-create "${WORKSPACE_NAME}" "${SERVICE_NAME}" "${SERVICE_PLAN_NAME}" "${REGION}" -g "${RESOURCE_GROUP}")
+
+      SERVICE_INSTANCE_ID=$(echo "$SERVICE_INSTANCE_OUTPUT" | grep -oE 'GUID:[[:space:]]+[^:[:space:]]+' | awk '{print $2}')
+
+      echo ${SERVICE_INSTANCE_ID}
+      # After the workspace is created, invoke the automation code
+      cd ${IBMCLOUD_HOME_FOLDER} && git clone -b release-${OCP_VERSION} https://github.com/IBM/ocp4-upi-compute-powervs.git
+
+      # Check if the terraform is of required version
+
+      # Populate the values in vars.tfvars
+      IC_API_KEY="$(< "${CLUSTER_PROFILE_DIR}/ibmcloud-api-key")"
+
+      # TODO: Should ssh keys ideally be copied from some secrets?
+      # copy public and private key files to the data directory
```

Comment:
create-var-file.sh https://github.com/IBM/ocp4-upi-compute-powervs/blob/main/scripts/create-var-file.sh does copy it to the data directory.